src/SFML/Graphics/Glsl.cpp

```diff
@@ -27,7 +27,6 @@
 // Headers
 ////////////////////////////////////////////////////////////
 #include <SFML/Graphics/Glsl.hpp>
-#include <algorithm>
```

Comment:
@vittorioromeo will love this :)

Reply:
Sorry, I went back to using `std::copy`!
I noticed there is `priv::copy` is `System/Utf.hpp`. Maybe this can be in a dedicated header to reuse it and avoid `<algorithm>`?

Reply:
If we're just copying contiguous memory from one place to another then `std::memcpy` will do the job quickly and without an expensive header inclusion. Right?

Reply:
```cpp
    for (std::size_t i = 0; i < length; ++i)
        std::memcpy(&contiguous[matrixSize * i], matrixArray[i].array.data(), matrixArray[i].array.size());
```
We might also be able to remove the `matrixSize` local variable since that's encoded `.array.size()`.

Reply:
Well, `std::memcpy` this is very much C, more error-prone, taking void pointers. Not sure we want that.

Reply:
> more error-prone

See, in your example above, `matrixArray[i].array.size()` should be `matrixArray[i].array.size() * sizeof(float)`

Reply:
Haha oops. Point taken. So `memcpy` isn't the way to go.

Reply:
Ultimately it depends how desperate we are to avoid `<algorithm>`. It is possible to get `std::memcpy` right even if I messed it up on my first attempt. We can make it look a little better like this:
```cpp
std::memcpy(&contiguous[matrixSize * i], matrixArray[i].array.data(), sizeof(matrixArray[i].array));
```
which should be correct since the sizeof a `std::array` is actually what we mean, not the number of elements.

Reply:
I think the C-style ugliness is not worth it, but let's see what others think.

Reply:
I think it's OK to have `<algorithm>` in .cpp files, the user compiles them only once (if at all) and doesn't pay for the extra overhead on each compilation.

The old `copyMatrix()` implementation has done its job well, I don't think we need to go out of our way to find a new implementation.